addons/l10n_vn_edi_viettel/models/account_move.py

```diff
@@ -0,0 +1,766 @@
+        compute=lambda self: self._compute_linked_attachment_id('l10n_vn_edi_sinvoice_xml_file_id', 'l10n_vn_edi_sinvoice_xml_file'),
+        depends=['l10n_vn_edi_sinvoice_xml_file'],
+        export_string_translation=False,
+    )
+    l10n_vn_edi_sinvoice_xml_file = fields.Binary(
+        string='SInvoice xml File',
+        copy=False,
+        export_string_translation=False,
+    )
+    l10n_vn_edi_sinvoice_pdf_file_id = fields.Many2one(
+        comodel_name='ir.attachment',
+        compute=lambda self: self._compute_linked_attachment_id('l10n_vn_edi_sinvoice_pdf_file_id', 'l10n_vn_edi_sinvoice_pdf_file'),
+        depends=['l10n_vn_edi_sinvoice_pdf_file'],
+        export_string_translation=False,
+    )
+    l10n_vn_edi_sinvoice_pdf_file = fields.Binary(
+        string='SInvoice pdf File',
+        copy=False,
+        export_string_translation=False,
+    )
+    # Replacement/Adjustment fields
+    l10n_vn_edi_agreement_document_name = fields.Char(
+        string='Agreement Name',
+        copy=False,
+    )
+    l10n_vn_edi_agreement_document_date = fields.Datetime(
+        string='Agreement Date',
+        copy=False,
+    )
+    l10n_vn_edi_adjustment_type = fields.Selection(
+        string='Adjustment type',
+        selection=[
+            ('1', 'Money adjustment'),
+            ('2', 'Information adjustment'),
+        ],
+        copy=False,
+    )
+    # Only used in case of replacement invoice.
+    l10n_vn_edi_replacement_origin = fields.Many2one(
+        comodel_name='account.move',
+        string='Replacement of',
+        copy=False,
+        readonly=True,
+        check_company=True,
+        export_string_translation=False,
+    )
+
+    @api.depends('l10n_vn_edi_invoice_state')
+    def _compute_show_reset_to_draft_button(self):
+        # EXTEND 'account'
+        super()._compute_show_reset_to_draft_button()
+        self.filtered(lambda m: m._l10n_vn_edi_is_sent() and m.l10n_vn_edi_invoice_state != 'canceled').show_reset_to_draft_button = False
+
+    @api.depends('l10n_vn_edi_invoice_state')
+    def _compute_need_cancel_request(self):
+        # EXTEND 'account' to add dependencies
+        return super()._compute_need_cancel_request()
+
+    @api.depends('payment_state')
+    def _compute_l10n_vn_edi_invoice_state(self):
+        """ Automatically set the state to payment_state_to_update when the payment state is updated.
+
+        This is a bit simplistic, as it can be wrongly set (for example, no need to send when going from in_payment to paid)
+        But this shouldn't be an issue since the logic to send the update will check if anything need to change.
+        """
+        for invoice in self.filtered(lambda i: i.country_code == 'VN'):
+            if invoice.l10n_vn_edi_invoice_state == 'sent':
+                invoice.l10n_vn_edi_invoice_state = 'payment_state_to_update'
+            else:
+                invoice.l10n_vn_edi_invoice_state = invoice.l10n_vn_edi_invoice_state
```

Comment:
Even if you fill a value only for VN, it remains a stored field and you must assign explicitely a value on all records
```suggestion
        for invoice in self:
            if invoice.country_code == 'VN' and invoice.l10n_vn_edi_invoice_state == 'sent':
                invoice.l10n_vn_edi_invoice_state = 'payment_state_to_update'
            else:
                invoice.l10n_vn_edi_invoice_state = invoice.l10n_vn_edi_invoice_state
```

FYI, `filtered`on a recordset is not lazy so:
```py
for record in self.filtered('turlutu'):
    ...
```
is less efficient than:
```py
for record in self:
    if not record.turlututu:
        continue
    ...
```